tests/test_layers.py

```diff
@@ -271,11 +271,19 @@ def model_func_2(wrapped_import, inputs: torch.Tensor):
     # test case (..., a, b*c,...) -> (..., a*b, c,...)
     return inputs.view(n_batch, n_time, n_feature_1 * 2, n_feature_2 // 2)
 
+  def model_func_3(wrapped_import, inputs: torch.Tensor):
+    # test case (..., a, b, F,...) -> (..., b, a*F,...)
+    inputs = inputs.transpose(1, 2).contiguous()
+    return inputs.view(n_batch, n_time, n_feature_1 * n_feature_2)
+
   rnd = numpy.random.RandomState(42)
   x = rnd.normal(0., 1., (n_batch, n_time, n_feature_1, n_feature_2)).astype("float32")
   for model_func in [model_func_1, model_func_2]:
     verify_torch_and_convert_to_returnn(model_func, inputs=x, inputs_data_kwargs={
         "shape": (None, n_feature_1, n_feature_2), "batch_dim_axis": 0, "time_dim_axis": 1, "feature_dim_axis": 2})
+  x = rnd.normal(0., 1., (n_batch, n_feature_1, n_time, n_feature_2)).astype("float32")
+  verify_torch_and_convert_to_returnn(model_func_3, inputs=x, inputs_data_kwargs={
```

Comment:
Normally, you should not extend a single test, but always make separate tests. It's already bad that we have `model_func_1` and `model_func_2` here. That should also better be separate. Every test case should be atomic if possible (esp if there are no real reasons not to do so).